builtin/logical/mssql/util.go

```diff
@@ -0,0 +1,60 @@
+package mssql
+
+import (
+	"fmt"
+	"strings"
+)
+
+// BuildDsn creates a DSN with some default and overriden values
+func BuildDsn(dsn string) string {
```

Comment:
This method is parsing all the parameters and re-joining them, to add `"app name" = "vault"`? If yes, can we remove this method altogether and append the `app name` to the connection string itself?

Is there a problem if the existing `app name` argument is overwritten to contain `vault`?